tests/configuration/pages/pmmInventoryPage.js

```diff
@@ -373,4 +455,130 @@ module.exports = {
     await I.waitForVisible(agent, 30);
     I.click(this.fields.backToServices);
   },
+
+  async clearFields() {
+    I.clearField(this.fields.environment);
+    I.clearField(this.fields.cluster);
+    I.clearField(this.fields.replicationSet);
+  },
+
+  async fillFields(serviceParameters) {
+    I.fillField(this.fields.environment, serviceParameters.environment);
+    I.fillField(this.fields.cluster, serviceParameters.cluster);
+    I.fillField(this.fields.replicationSet, serviceParameters.replicationSet);
+  },
+
+  async saveConfirm() {
+    I.click(this.fields.saveButton);
+    I.seeElement(this.fields.savePopupMessage);
+    I.click(this.fields.saveConfirmButton);
+  },
+
+  async checkLabels(serviceParameters) {
+    var labels;
+    labels = `//span[contains(text(),"${serviceParameters.environment}")]`;
+    I.waitForElement(labels, 30);
+    labels = `//span[contains(text(),"${serviceParameters.cluster}")]`;
+    I.waitForElement(labels, 30);
+    labels = `//span[contains(text(),"${serviceParameters.replicationSet}")]`;
+    I.waitForElement(labels, 30);
+  },
+  
+  async editRemoteServiceDisplayed(serviceName) {
+    I.waitForElement(this.fields.kebabMenu(serviceName),30);
+    I.wait(10);
+    I.click(this.fields.kebabMenu(serviceName));
+    I.waitForElement(this.fields.editAction,30);
+    I.wait(10);
+    I.click(this.fields.editAction);
+    I.waitForElement(this.fields.editText);
+    I.seeElement(this.fields.editText);
+    switch (serviceName) {
```

Comment:
please remove this long hardcoded switch-case block in favor of generic function argument with doc: functions should accept object of a definite structure eg: `{ serviceName: string, settingsForm: {[formFiled: string]: string } }`

Reply:
Thanks, addressed